### DISCUSSION TITLE
Multiple `Session.AddHandler` for the same event

### DISCUSSION THREAD

**Discussion by Malix-Labs** [bwmarrin/discordgo#1446] [Q&A] [answered]
Using [`Session.AddHandler`](https://pkg.go.dev/github.com/bwmarrin/discordgo#Session.AddHandler), is there a difference between calling it once, and calling it multiple times with each having the same event parameter?
Should we only call it once per event?

**Comment by Earlopain** [comment, marked as answer]
Event handlers are saved internally in a `map[string][]*eventHandlerInstance{}` where `string` is the event handler type. Registering event handlers for the same event multiple times is not an issue.
